src/engine/strat_engine/cmd.rs

```diff
@@ -70,16 +70,16 @@ pub fn verify_binaries() -> StratisResult<()> {
 
 /// Common function to call a command line utility, returning an Result with an error message which
 /// also includes stdout & stderr if it fails.
-fn execute_cmd(cmd: &mut Command, error_msg: &str) -> StratisResult<()> {
+fn execute_cmd(cmd: &mut Command) -> StratisResult<()> {
     let result = cmd.output()?;
```

Comment:
I think it would be better if we tested for error here instead of returning it, so that the same additional command information can be added to the error before returning it.

Reply:
Something like this:

```
use std::process::Command;

fn execute_cmd(cmd: &mut Command) -> Result<(), String> {
    match cmd.output() {
        Err(err) => {
            Err(format!("Failed to execute command {:?}, err: {:?}", cmd, err))
        }
        Ok(result) => {
            if result.status.success() {
                Ok(())
            } else {
                let std_out_txt = String::from_utf8_lossy(&result.stdout);
                let std_err_txt = String::from_utf8_lossy(&result.stderr);
                let err_msg = format!(
                    "cmd: {:?}, stdout: {} stderr: {}",
                    cmd, std_out_txt, std_err_txt
                );
                Err(err_msg)
            }
        }
    }
}

pub fn main() {
    execute_cmd(
        Command::new("junkfs")
            .arg("-f")
            .arg("-q")
            .arg("/dev/sda")
            .arg("-m"),
    ).unwrap();
}
```

would sort of work.

Reply:
:+1: